scripts/lib/CIME/case_st_archive.py

```diff
@@ -117,36 +209,41 @@ def _archive_rpointer_files(case, archive, archive_entry, archive_restdir,
             for rpointer_file, rpointer_content in rpointer_items:
                 temp_rpointer_file = rpointer_file
                 temp_rpointer_content = rpointer_content
+
                 # put in a temporary setting for ninst_strings if they are empty
                 # in order to have just one loop over ninst_strings below
                 if rpointer_content != 'unset':
                     if not ninst_strings:
                         ninst_strings = ["empty"]
-                    for ninst_string in ninst_strings:
-                        rpointer_file = temp_rpointer_file
-                        rpointer_content = temp_rpointer_content
-                        if ninst_string == 'empty':
-                            ninst_string = ""
-                        for key, value in [('$CASE', casename),
-                                           ('$DATENAME', datename),
-                                           ('$NINST_STRING', ninst_string)]:
-                            rpointer_file = rpointer_file.replace(key, value)
-                            rpointer_content = rpointer_content.replace(key, value)
-
-                        # write out the respective files with the correct contents
-                        rpointer_file = os.path.join(archive_restdir, rpointer_file)
-                        logger.debug("writing rpointer_file {}".format(rpointer_file))
-                        f = open(rpointer_file, 'w')
-                        for output in rpointer_content.split(','):
-                            f.write("{} \n".format(output))
-                        f.close()
+
+                for ninst_string in ninst_strings:
```

Comment:
This is out-dented a level relative to the original. I have a vague recollection that the indenting an extra level was a bug fix put in by @kdraeder  (so this may be a regression).